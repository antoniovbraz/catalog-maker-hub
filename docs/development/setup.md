# Configuração do Ambiente

## Instalação

1. Certifique-se de ter o Node.js 18 ou superior instalado.
2. Instale as dependências com npm:

```bash
npm install
```

## Execução

Inicie o servidor de desenvolvimento com:

```bash
npm run dev
```

## Supabase CLI

1. Instale a CLI:

   ```bash
   npm install -g supabase
   ```

2. Inicie os serviços locais:

   ```bash
   supabase start
   ```

3. Aplique as migrações existentes:

   ```bash
   supabase db reset
   ```

4. Configure os secrets necessários:

   ```bash
   cp supabase/.env.example supabase/.env
   supabase secrets set --env-file supabase/.env
   ```

5. Valide o schema com o documento [database-schema.md](database-schema.md).

## Verificações de Qualidade

Execute todas as verificações antes de abrir um pull request:

```bash
npm run lint
npm run lint:fix
npm run type-check
npm test
npm run test:coverage
```

## Referências de Documentação

- Escreva a documentação em português.
- Utilize o estilo descrito em [docs/README.md](../README.md).
- Prefira links relativos e use `https://peepers-hub.lovable.app` para URLs de produção.

