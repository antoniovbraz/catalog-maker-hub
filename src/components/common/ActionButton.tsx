import { ReactNode } from "react";
import { Button } from "@/components/ui/button";
import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from "@/components/ui/tooltip";
import { cn } from "@/lib/utils";

interface ActionButtonProps {
  icon: ReactNode;
  label: string;
  onClick: () => void;
  variant?: "default" | "destructive" | "outline" | "secondary" | "ghost" | "link";
  size?: "default" | "sm" | "lg" | "icon";
  disabled?: boolean;
  loading?: boolean;
  tooltip?: string;
  className?: string;
}

export function ActionButton({
  icon,
  label,
  onClick,
  variant = "outline",
  size = "sm",
  disabled = false,
  loading = false,
  tooltip,
  className
}: ActionButtonProps) {
  const button = (
    <Button
      variant={variant}
      size={size}
      onClick={onClick}
      disabled={disabled || loading}
      className={cn(
        "flex items-center gap-2 transition-all duration-200",
        "hover:scale-[1.02] active:scale-[0.98]",
        loading && "cursor-wait",
        className
      )}
    >
      <span className={cn(
        "transition-transform duration-200",
        loading && "animate-spin"
      )}>
        {icon}
      </span>
      {size !== "icon" && (
        <span className="font-medium">
          {loading ? "Processando..." : label}
        </span>
      )}
    </Button>
  );

  if (tooltip) {
    return (
      <TooltipProvider>
        <Tooltip>
          <TooltipTrigger asChild>
            {button}
          </TooltipTrigger>
          <TooltipContent>
            <p className="text-sm">{tooltip}</p>
          </TooltipContent>
        </Tooltip>
      </TooltipProvider>
    );
  }

  return button;
}