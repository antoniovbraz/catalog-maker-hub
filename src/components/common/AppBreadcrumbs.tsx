import { Fragment } from "react";
import { Home } from '@/components/ui/icons';
import { Link, useLocation } from "react-router-dom";
import { cn } from "@/lib/utils";
import {
  Breadcrumb,
  BreadcrumbList,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbPage,
  BreadcrumbSeparator,
} from "@/components/ui/breadcrumb";

interface BreadcrumbEntry {
  label: string;
  path?: string;
}

interface AppBreadcrumbsProps {
  items?: BreadcrumbEntry[];
  className?: string;
}

const routeLabels: Record<string, string> = {
  "/dashboard": "Dashboard",
  "/strategy": "Estratégia",
  "/marketplaces": "Marketplaces",
  "/categories": "Categorias",
  "/products": "Produtos",
  "/shipping": "Frete",
  "/commissions": "Comissões",
  "/fixed-fees": "Taxas Fixas",
  "/sales": "Vendas",
  "/pricing": "Precificação",
  "/subscription": "Assinaturas",
  "/admin": "Dashboard Admin",
};

export function AppBreadcrumbs({ items, className }: AppBreadcrumbsProps) {
  const location = useLocation();

  // Generate items automatically when none are provided
  const breadcrumbItems =
    items ||
    (() => {
      const pathSegments = location.pathname.split("/").filter(Boolean);
      const generated: BreadcrumbEntry[] = [
        { label: "Home", path: "/dashboard" },
      ];

      let currentPath = "";
      pathSegments.forEach((segment) => {
        currentPath += `/${segment}`;
        const label =
          routeLabels[currentPath] ||
          segment.charAt(0).toUpperCase() + segment.slice(1);
        generated.push({ label, path: currentPath });
      });

      return generated;
    })();

  // Hide when only home is present
  if (breadcrumbItems.length <= 1) {
    return null;
  }

  return (
    <Breadcrumb className={cn(className)}>
      <BreadcrumbList>
        <BreadcrumbItem>
          <BreadcrumbLink asChild>
            <Link to="/dashboard" className="inline-flex items-center gap-1">
              <Home className="size-4" />
              <span className="sr-only">Dashboard</span>
            </Link>
          </BreadcrumbLink>
        </BreadcrumbItem>

        {breadcrumbItems.slice(1).map((item, index) => {
          const isLast = index === breadcrumbItems.length - 2;
          return (
            <Fragment key={item.path || item.label}>
              <BreadcrumbSeparator />
              <BreadcrumbItem>
                {isLast ? (
                  <BreadcrumbPage>{item.label}</BreadcrumbPage>
                ) : (
                  <BreadcrumbLink asChild>
                    <Link to={item.path || "#"}>{item.label}</Link>
                  </BreadcrumbLink>
                )}
              </BreadcrumbItem>
            </Fragment>
          );
        })}
      </BreadcrumbList>
    </Breadcrumb>
  );
}

export default AppBreadcrumbs;
