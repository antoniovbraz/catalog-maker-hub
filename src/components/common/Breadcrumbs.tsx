import { ChevronRight, Home } from "lucide-react";
import { Link, useLocation } from "react-router-dom";
import { cn } from "@/lib/utils";

interface BreadcrumbItem {
  label: string;
  path?: string;
}

interface BreadcrumbsProps {
  items?: BreadcrumbItem[];
  className?: string;
}

const routeLabels: Record<string, string> = {
  '/dashboard': 'Dashboard',
  '/strategy': 'Estratégia',
  '/marketplaces': 'Marketplaces',
  '/categories': 'Categorias',
  '/products': 'Produtos',
  '/shipping': 'Frete',
  '/commissions': 'Comissões',
  '/fixed-fees': 'Taxas Fixas',
  '/sales': 'Vendas',
  '/pricing': 'Precificação',
  '/subscription': 'Assinaturas',
  '/admin': 'Dashboard Admin',
};

export function Breadcrumbs({ items, className }: BreadcrumbsProps) {
  const location = useLocation();
  
  // Se não há items fornecidos, gerar automaticamente baseado na rota
  const breadcrumbItems = items || (() => {
    const pathSegments = location.pathname.split('/').filter(Boolean);
    const generatedItems: BreadcrumbItem[] = [{ label: 'Home', path: '/dashboard' }];
    
    let currentPath = '';
    pathSegments.forEach((segment) => {
      currentPath += `/${segment}`;
      const label = routeLabels[currentPath] || segment.charAt(0).toUpperCase() + segment.slice(1);
      generatedItems.push({
        label,
        path: currentPath
      });
    });
    
    return generatedItems;
  })();

  // Se só tem um item (Home), não mostrar breadcrumbs
  if (breadcrumbItems.length <= 1) {
    return null;
  }

  return (
    <nav className={cn("flex items-center space-x-1 text-sm text-muted-foreground", className)}>
      <Link 
        to="/dashboard" 
        className="flex items-center hover:text-foreground transition-colors"
      >
        <Home className="w-4 h-4" />
      </Link>
      
      {breadcrumbItems.slice(1).map((item, index) => {
        const isLast = index === breadcrumbItems.length - 2;
        
        return (
          <div key={item.path || item.label} className="flex items-center">
            <ChevronRight className="w-4 h-4 mx-1" />
            {isLast ? (
              <span className="font-medium text-foreground">{item.label}</span>
            ) : (
              <Link 
                to={item.path || '#'} 
                className="hover:text-foreground transition-colors"
              >
                {item.label}
              </Link>
            )}
          </div>
        );
      })}
    </nav>
  );
}