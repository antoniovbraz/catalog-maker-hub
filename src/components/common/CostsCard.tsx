import React from "react";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { Tag, Info } from "@/components/ui/icons";
import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from "@/components/ui/tooltip";
import { formatarMoeda } from "@/utils/pricing";
import type { ProductWithCategory } from "@/types/products";

interface ProductPriceInfo {
  price: number;
  source: 'ml' | 'product';
}

interface CostsCardProps {
  product: ProductWithCategory;
  priceInfo?: ProductPriceInfo | null;
}

export function CostsCard({ product, priceInfo }: CostsCardProps): React.ReactElement {
  return (
    <Card>
      <CardHeader>
        <CardTitle className="flex items-center gap-2">
          <Tag className="size-5" />
          Custos e Impostos
        </CardTitle>
      </CardHeader>
      <CardContent>
        <div className="grid grid-cols-4 gap-4">
          <div>
            <div className="flex items-center gap-1">
              <label className="text-sm font-medium text-muted-foreground">Preço de venda</label>
              {priceInfo?.source === "ml" && (
                <TooltipProvider>
                  <Tooltip>
                    <TooltipTrigger asChild>
                      <Info className="size-3 text-muted-foreground" />
                    </TooltipTrigger>
                    <TooltipContent>Preço do Mercado Livre</TooltipContent>
                  </Tooltip>
                </TooltipProvider>
              )}
            </div>
            <p className="text-lg font-semibold">{formatarMoeda(priceInfo?.price || 0)}</p>
          </div>
          <div>
            <label className="text-sm font-medium text-muted-foreground">Custo Unitário</label>
            <p className="text-lg font-semibold">{formatarMoeda(product.cost_unit)}</p>
          </div>
          <div>
            <label className="text-sm font-medium text-muted-foreground">Custo Embalagem</label>
            <p className="text-lg font-semibold">{formatarMoeda(product.packaging_cost || 0)}</p>
          </div>
          <div>
            <label className="text-sm font-medium text-muted-foreground">Taxa de Imposto</label>
            <p className="text-lg font-semibold">{product.tax_rate || 0}%</p>
          </div>
        </div>
      </CardContent>
    </Card>
  );
}