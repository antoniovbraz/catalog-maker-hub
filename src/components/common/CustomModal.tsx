import * as React from "react";
import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogFooter,
  DialogHeader,
  DialogTitle,
} from "@/components/ui/dialog";
import { Button } from "@/components/ui/button";
import { cn } from "@/lib/utils";

interface CustomModalProps {
  open: boolean;
  onOpenChange: (open: boolean) => void;
  title: string;
  description?: string;
  children: React.ReactNode;
  onSave: () => void;
  onCancel?: () => void;
  saveText?: string;
  cancelText?: string;
  isLoading?: boolean;
  className?: string;
  size?: "sm" | "md" | "lg" | "xl";
}

export function CustomModal({
  open,
  onOpenChange,
  title,
  description,
  children,
  onSave,
  onCancel,
  saveText = "Salvar",
  cancelText = "Cancelar",
  isLoading = false,
  className,
  size = "md",
}: CustomModalProps) {
  const handleCancel = () => {
    if (onCancel) {
      onCancel();
    } else {
      onOpenChange(false);
    }
  };

  const sizeClasses = {
    sm: "max-w-md",
    md: "max-w-lg",
    lg: "max-w-2xl", 
    xl: "max-w-4xl",
  };

  return (
    <Dialog open={open} onOpenChange={onOpenChange}>
      <DialogContent 
        className={cn(
          "gap-0 p-0 overflow-hidden",
          sizeClasses[size],
          className
        )}
      >
        {/* Header */}
        <DialogHeader className="px-6 py-5 border-b bg-background">
          <DialogTitle className="text-xl font-semibold text-foreground">
            {title}
          </DialogTitle>
          {description && (
            <DialogDescription className="text-sm text-muted-foreground mt-1">
              {description}
            </DialogDescription>
          )}
        </DialogHeader>

        {/* Content */}
        <div className="px-6 py-6 max-h-[70vh] overflow-y-auto">
          {children}
        </div>

        {/* Footer */}
        <DialogFooter className="px-6 py-4 border-t bg-muted/30 flex justify-end gap-3">
          <Button
            type="button"
            variant="outline"
            onClick={handleCancel}
            disabled={isLoading}
            className="min-w-[80px]"
          >
            {cancelText}
          </Button>
          <Button
            type="button"
            onClick={onSave}
            disabled={isLoading}
            className="min-w-[80px] bg-foreground text-background hover:bg-foreground/90"
          >
            {isLoading ? "Salvando..." : saveText}
          </Button>
        </DialogFooter>
      </DialogContent>
    </Dialog>
  );
}