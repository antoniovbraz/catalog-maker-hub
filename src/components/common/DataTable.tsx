import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from "@/components/ui/table";
import { Button } from "@/components/ui/button";
import { Badge } from "@/components/ui/badge";
import { Edit, Trash2 } from "lucide-react";
import { EmptyState } from "./EmptyState";
import { LoadingSpinner } from "./LoadingSpinner";

export interface Column<T> {
  key: keyof T | string;
  header: string;
  render?: (value: unknown, item: T) => React.ReactNode;
  className?: string;
}

export interface DataTableProps<T> {
  data: T[];
  columns: Column<T>[];
  onEdit?: (item: T) => void;
  onDelete?: (item: T) => void;
  loading?: boolean;
  emptyMessage?: string;
  emptyDescription?: string;
}

export function DataTable<T extends { id: string }>({
  data,
  columns,
  onEdit,
  onDelete,
  loading = false,
  emptyMessage = "Nenhum registro encontrado",
  emptyDescription = "Adicione o primeiro registro para começar",
}: DataTableProps<T>) {
  if (loading) {
    return <LoadingSpinner />;
  }

  if (!data.length) {
    return <EmptyState message={emptyMessage} description={emptyDescription} />;
  }

  const getValue = (item: T, key: string) => {
    return key.includes('.') 
      ? key.split('.').reduce((obj, k) => obj?.[k], item as any)
      : (item as any)[key];
  };

  return (
    <div className="rounded-md border">
      <Table>
        <TableHeader>
          <TableRow>
            {columns.map((column, index) => (
              <TableHead key={index} className={column.className}>
                {column.header}
              </TableHead>
            ))}
            {(onEdit || onDelete) && (
              <TableHead className="text-right">Ações</TableHead>
            )}
          </TableRow>
        </TableHeader>
        <TableBody>
          {data.map((item, index) => (
            <TableRow key={item.id || index}>
              {columns.map((column, colIndex) => {
                const value = getValue(item, column.key as string);
                return (
                  <TableCell key={colIndex} className={column.className}>
                    {column.render ? column.render(value, item) : value}
                  </TableCell>
                );
              })}
              {(onEdit || onDelete) && (
                <TableCell className="text-right">
                  <div className="flex justify-end gap-2">
                    {onEdit && (
                      <Button
                        variant="ghost"
                        size="sm"
                        onClick={() => onEdit(item)}
                      >
                        <Edit className="h-4 w-4" />
                      </Button>
                    )}
                    {onDelete && (
                      <Button
                        variant="ghost"
                        size="sm"
                        onClick={() => onDelete(item)}
                      >
                        <Trash2 className="h-4 w-4" />
                      </Button>
                    )}
                  </div>
                </TableCell>
              )}
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </div>
  );
}