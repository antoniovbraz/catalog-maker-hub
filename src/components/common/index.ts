export { LoadingSpinner } from "./LoadingSpinner"
export { ConfirmDialog } from "./ConfirmDialog"
export { FormField } from "./FormField"
export { MiniProgressBar } from "./MiniProgressBar"
export { EnhancedTooltip } from "./EnhancedTooltip"
export { AppBreadcrumbs } from "./AppBreadcrumbs"
export { SkeletonTable, SkeletonForm, SkeletonCard } from "./SkeletonLoaders"
export { ActionButton } from "./ActionButton"
export { PageTransition } from "./PageTransition"
