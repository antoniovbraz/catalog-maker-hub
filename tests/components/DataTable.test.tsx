import { describe, it, expect, vi, beforeEach } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { testUtils } from '../setup';
import { DataTable } from '@/components/common/DataTable';

// Mock do React Router
vi.mock('react-router-dom', () => ({
  useNavigate: () => vi.fn(),
}));

const createWrapper = () => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: { retry: false },
      mutations: { retry: false },
    },
  });
  
  return ({ children }: { children: React.ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      {children}
    </QueryClientProvider>
  );
};

describe('DataTable Component', () => {
  beforeEach(() => {
    testUtils.resetAllMocks();
  });

  const mockData = [
    { id: '1', name: 'Item 1', status: 'active' },
    { id: '2', name: 'Item 2', status: 'inactive' },
  ];

  const mockColumns = [
    { key: 'name', header: 'Nome' },
    { key: 'status', header: 'Status' },
  ];

  it('deve renderizar tabela com dados', () => {
    render(
      <DataTable 
        data={mockData} 
        columns={mockColumns} 
      />, 
      { wrapper: createWrapper() }
    );

    expect(screen.getByText('Nome')).toBeInTheDocument();
    expect(screen.getByText('Status')).toBeInTheDocument();
    expect(screen.getByText('Item 1')).toBeInTheDocument();
    expect(screen.getByText('Item 2')).toBeInTheDocument();
    expect(screen.getByText('active')).toBeInTheDocument();
    expect(screen.getByText('inactive')).toBeInTheDocument();
  });

  it('deve renderizar estado de loading', () => {
    render(
      <DataTable 
        data={[]} 
        columns={mockColumns} 
        loading={true}
      />, 
      { wrapper: createWrapper() }
    );

    expect(screen.getByTestId('loading-spinner')).toBeInTheDocument();
  });

  it('deve renderizar estado vazio', () => {
    render(
      <DataTable 
        data={[]} 
        columns={mockColumns}
        emptyMessage="Nenhum item encontrado"
        emptyDescription="Adicione um item para começar"
      />, 
      { wrapper: createWrapper() }
    );

    expect(screen.getByText('Nenhum item encontrado')).toBeInTheDocument();
    expect(screen.getByText('Adicione um item para começar')).toBeInTheDocument();
  });

  it('deve renderizar botões de ação quando fornecidos', () => {
    const mockOnEdit = vi.fn();
    const mockOnDelete = vi.fn();

    render(
      <DataTable 
        data={mockData} 
        columns={mockColumns}
        onEdit={mockOnEdit}
        onDelete={mockOnDelete}
      />, 
      { wrapper: createWrapper() }
    );

    const editButtons = screen.getAllByRole('button');
    expect(editButtons).toHaveLength(4); // 2 edit + 2 delete
  });

  it('deve chamar callback de edição quando botão é clicado', async () => {
    const mockOnEdit = vi.fn();
    const user = userEvent.setup();

    render(
      <DataTable 
        data={mockData} 
        columns={mockColumns}
        onEdit={mockOnEdit}
      />, 
      { wrapper: createWrapper() }
    );

    const editButtons = screen.getAllByRole('button');
    await user.click(editButtons[0]); // Primeiro botão de editar

    expect(mockOnEdit).toHaveBeenCalledWith(mockData[0]);
  });

  it('deve chamar callback de deleção quando botão é clicado', async () => {
    const mockOnDelete = vi.fn();
    const user = userEvent.setup();

    render(
      <DataTable 
        data={mockData} 
        columns={mockColumns}
        onDelete={mockOnDelete}
      />, 
      { wrapper: createWrapper() }
    );

    const deleteButtons = screen.getAllByRole('button');
    const deleteButton = deleteButtons.find(button => 
      button.querySelector('svg')?.getAttribute('data-testid') === 'trash-2'
    );
    
    if (deleteButton) {
      await user.click(deleteButton);
      expect(mockOnDelete).toHaveBeenCalledWith(mockData[0]);
    }
  });

  it('deve renderizar conteúdo customizado com render function', () => {
    const customColumns = [
      {
        key: 'name',
        header: 'Nome'
      },
      {
        key: 'status',
        header: 'Status',
        render: (value) => {
          const status = value as string;
          return (
            <span className={status === 'active' ? 'text-green-500' : 'text-red-500'}>
              {status.toUpperCase()}
            </span>
          );
        }
      },
    ];

    render(
      <DataTable 
        data={mockData} 
        columns={customColumns}
      />, 
      { wrapper: createWrapper() }
    );

    expect(screen.getByText('ACTIVE')).toBeInTheDocument();
    expect(screen.getByText('INACTIVE')).toBeInTheDocument();
  });

  it('deve lidar com dados aninhados usando dot notation', () => {
    const nestedData = [
      { id: '1', user: { name: 'João' }, category: { name: 'Eletrônicos' } },
      { id: '2', user: { name: 'Maria' }, category: { name: 'Casa' } },
    ];

    const nestedColumns = [
      { key: 'user.name', header: 'Usuário' },
      { key: 'category.name', header: 'Categoria' },
    ];

    render(
      <DataTable 
        data={nestedData} 
        columns={nestedColumns}
      />, 
      { wrapper: createWrapper() }
    );

    expect(screen.getByText('João')).toBeInTheDocument();
    expect(screen.getByText('Maria')).toBeInTheDocument();
    expect(screen.getByText('Eletrônicos')).toBeInTheDocument();
    expect(screen.getByText('Casa')).toBeInTheDocument();
  });

  it('deve corresponder ao snapshot', () => {
    const { container } = render(
      <DataTable
        data={mockData}
        columns={mockColumns}
      />,
      { wrapper: createWrapper() }
    );

    expect(container).toMatchSnapshot();
  });
});
