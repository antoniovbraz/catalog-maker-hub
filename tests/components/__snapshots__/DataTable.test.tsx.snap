// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataTable Component > deve corresponder ao snapshot 1`] = `
<div>
  <div
    class="space-y-4"
  >
    <div
      class="flex items-center justify-between"
    >
      <div
        class="flex items-center gap-3"
      >
        <div
          class="relative w-64"
        >
          <svg
            class="lucide lucide-search absolute left-3 top-1/2 transform -translate-y-1/2 text-muted-foreground w-4 h-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <path
              d="m21 21-4.3-4.3"
            />
          </svg>
          <input
            class="flex h-10 w-full rounded-md border-input px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm pl-10 bg-muted/50 border-0 focus-visible:ring-1 focus-visible:ring-primary"
            placeholder="Buscar..."
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="rounded-lg border border-border/50 bg-card shadow-card overflow-hidden"
    >
      <div
        class="relative w-full overflow-auto"
      >
        <table
          class="w-full caption-bottom text-sm"
        >
          <thead
            class="[&_tr]:border-b"
          >
            <tr
              class="border-b transition-colors data-[state=selected]:bg-muted bg-muted/50 hover:bg-muted/60"
            >
              <th
                class="h-12 px-4 text-left align-middle text-muted-foreground [&:has([role=checkbox])]:pr-0 font-semibold"
              >
                Nome
              </th>
              <th
                class="h-12 px-4 text-left align-middle text-muted-foreground [&:has([role=checkbox])]:pr-0 font-semibold"
              >
                Status
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
          >
            <tr
              class="border-b data-[state=selected]:bg-muted hover:bg-muted/30 transition-colors border-border/50"
            >
              <td
                class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
              >
                Item 1
              </td>
              <td
                class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
              >
                active
              </td>
            </tr>
            <tr
              class="border-b data-[state=selected]:bg-muted hover:bg-muted/30 transition-colors border-border/50"
            >
              <td
                class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
              >
                Item 2
              </td>
              <td
                class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
              >
                inactive
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="text-sm text-muted-foreground"
    >
      Mostrando 
      2
       de 
      2
       registros
    </div>
  </div>
</div>
`;
